src/metabase/driver/sql/references.clj

```diff
@@ -314,26 +346,30 @@
 
 (defmethod field-references-impl [:sql :macaw.ast/set-operation]
   [driver outside-sources withs expr]
-  {:used-fields (->> (find-used-fields driver outside-sources withs expr)
-                     (into #{}))
-   :returned-fields (->> (find-returned-fields driver outside-sources withs expr)
-                         (into []))
-   :names (when-let [alias (:table-alias expr)]
-            {:table-alias (sql.normalize/normalize-name driver alias)})})
+  (let [used-fields (find-used-fields driver outside-sources withs expr)
+        returned-fields (find-returned-fields driver outside-sources withs expr)]
+    {:used-fields (into #{} (keep :col) used-fields)
+     :returned-fields (into [] (keep :col) returned-fields)
+     :errors (-> #{}
+                 (into (mapcat :errors) used-fields)
+                 (into (mapcat :errors) returned-fields))
+     :names (when-let [alias (:table-alias expr)]
+              {:table-alias (sql.normalize/normalize-name driver alias)})}))
 
 (defmethod field-references-impl [:sql :macaw.ast/table-function]
   [driver _outside-sources _withs expr]
   {:used-fields #{}
    :returned-fields [{:type :unknown-columns}]
    :names (when-let [alias (:table-alias expr)]
-            {:table-alias (sql.normalize/normalize-name driver alias)})})
+            {:table-alias (sql.normalize/normalize-name driver alias)})
+   :errors #{}})
 
 (defmethod field-references-impl :default
   [_driver _outside-sources _outside-withs _expr]
   {:used-fields #{}
    :returned-fields []
    :names nil
-   :bad-sql true})
```

Comment:
I'm like 99% sure I suggested giving this these error keywords namespaced names in the last dependencies validation PR I reviewed, well I guess I'm just glad it's happening now. Sorta need to follow up and try to wrap my head around why it wasn't done way back then I guess 😞